lib/usd/ui/layerEditor/mayaCommandHook.cpp

```diff
@@ -295,4 +283,50 @@ bool MayaCommandHook::isProxyShapeSharedStage(const std::string& proxyShapePath)
     return getBooleanAttributeOnProxyShape(proxyShapePath, "shareStage");
 }
 
+std::string MayaCommandHook::executeMel(const std::string& commandString)
+{
+    if (areCommandsDelayed()) {
+        _delayedCommands.push_back({ commandString, false });
+    } else {
+        // executes maya command with display and undo set to true so that it logs
+        MStringArray result;
+        MGlobal::executeCommand(
+            MString(commandString.c_str()),
+            result,
+            /*display*/ true,
+            /*undo*/ true);
+        if (result.length() > 0)
+            return result[0].asChar();
+    }
+    return "";
+}
+
+void MayaCommandHook::executePython(const std::string& commandString)
+{
+    if (areCommandsDelayed()) {
+        _delayedCommands.push_back({ commandString, true });
+    } else {
+        MGlobal::executePythonCommand(commandString.c_str());
+    }
+}
+
+void MayaCommandHook::executeDelayedCommands()
```

Comment:
Would you mind adding a comment that this function should not be called directly?
Also, do we always want to have display on?